### PR TITLE
Do not canonicalize golden file if error != 0 for jsonnet_to_json_test.

### LINKED ISSUE #20: jsonnet_to_json_test should not canonicalize golden file if error != 0
+@sparkprime

Just realized that with the changes in #16, `jsonnet_to_json_test`s for negative tests (i.e. `error` set to non-zero) are failing because we run `jsonnet` on the golden file, which in this case contains the expected error output.

The fix for this is to have the generated shell script cat the golden file if `ctx.attr.error != 0` and use the default behavior of using `jsonnet` to canonicalize the golden file otherwise.

### PR DESCRIPTION
**Pull request by davidzchen**
Fixes #20

### CODE REVIEW

**Comment by davidzchen** [comment]
+@sparkprime

File: jsonnet/jsonnet.bzl [hunk 1]
```diff
@@ -180,15 +180,18 @@ def _jsonnet_to_json_test_impl(ctx):
   diff_command = ""
   if ctx.file.golden:
     golden_files += [ctx.file.golden]
+    # Note that we only run jsonnet to canonicalize the golden output if the
+    # expected return code is 0. Otherwise, the golden file contains the
+    # expected error output.
     if ctx.attr.regex:
       diff_command = _REGEX_DIFF_COMMAND % (
-          ctx.file.jsonnet.short_path,
+          ctx.file.jsonnet.short_path if ctx.attr.error == 0 else 'cat',
```
Comment (damienmg) [inline jsonnet/jsonnet.bzl]: This made the change hard to understand. jssonnet is an  executable right? In that case, I recommend using ctx.executable just to make it clear that you just using a different command.

(also for the other diff command)

Comment (davidzchen) [inline jsonnet/jsonnet.bzl]: Done.

**Comment by davidzchen** [comment]
The background for this is that bazelbuild/rules_jsonnet#16 made a change where we would run `jsonnet` on the golden files as part of using `jsonnet_to_json_test` for golden tests in order to normalize the style of both the expected and actual outputs. However, when one uses `jsonnet_to_json_test` to diff against expected error output, running `jsonnet` on the error output results in a different error, and as a result, this change `cat`s the golden file instead in this case.

**Pull request merged**